easy_transformer/hook_points.py

```diff
@@ -1,149 +1,216 @@
-# Import stuff
 import logging
-from typing import Callable, Union, Optional, Sequence
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Callable, Optional, Sequence, Union
+
 import torch
 import torch.nn as nn
 
-from functools import *
 
-# %%
-# Define type aliases
-NamesFilter = Optional[Union[Callable[[str], bool], Sequence[str]]]
+class NamesFilter(ABC):
+    @abstractmethod
+    def is_included(self, name: str) -> bool:
+        pass
+
+
+class FunctionalNameFilter(NamesFilter):
+    def __init__(self, f: Callable[[str], bool]):
+        self.f = f
+
+    def is_included(self, name: str) -> bool:
+        return self.f(name)
+
+
+class ConstantNameFilter(NamesFilter):
```

Comment:
There should also be a name class for just inputting a single name (Which is actually the standard use case lol)

Reply:
d17cff8